content/docs/guides/converting-gatsby-to-tina.mdx

```diff
@@ -0,0 +1,464 @@
+  const { createNodeField } = actions
+  if (node.internal.type === `Mdx`) {
+    const value = createFilePath({ node, getNode })
++   createNodeField({
++     name: `tinaMarkdown`,
++     node,
++     value: JSON.stringify(parseMDX(node.body)),
++   })
+    createNodeField({
+      name: `slug`,
+      node,
+      value,
+    })
+  }
+}
+```
+
+There's still a few parameters we need to provide to `parseMDX` before it can do it's job.
+
+- The second argument is a copy of the field we defined in our `config.js` file.
+- The third is a callback function that the parser uses when coming across any image links. We'll use that callback function to update our image urls to point to the image directory we defined in our config file.
+
+```diff
++ const imageCallback = url => {
++   return url.replace(/^\./, "/images")
++ }
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+  if (node.internal.type === `Mdx`) {
+    const value = createFilePath({ node, getNode })
+    createNodeField({
+      name: `tinaMarkdown`,
+      node,
+      value: JSON.stringify(
+        parseMDX(
+          node.body,
++          {
++           type: "rich-text",
++           name: "body",
++           label: "Body",
++           isBody: true,
++           },
++           imageCallback
+        )
+      ),
+    })
+// ...
+```
+
+Now that we've defined our new custom node we should be ready to use it in our graphql query. Open `src/templates/blog-post.js` and add our custom node to the page query.
+
+```
+export const pageQuery = graphql`
+  query BlogPostBySlug(
+    $id: String!
+    $previousPostId: String
+    $nextPostId: String
+  ) {
+    site {
+      siteMetadata {
+        title
+      }
+    }
+    mdx(id: { eq: $id }) {
+      id
++     fields {
++       tinaMarkdown
++     }
+// ...
+`
+```
+
+Now we're ready to parse the formatted markdown into our component. We'll also remove the `"children"` prop which normally contains the HTML from Gatsby's mdx parser.
+
+```diff
+const BlogPostTemplate = ({
+  data: { previous, next, site, mdx: post },
+  location,
+- children,
+}) => {
+  const siteTitle = site.siteMetadata?.title || `Title`
++ const tinaMarkdownContent = JSON.parse(post.fields.tinaMarkdown)
+  return (
+    <Layout location={location} title={siteTitle}>
+      <article
+        className="blog-post"
+        itemScope
+        itemType="http://schema.org/Article"
+      >
+        <header>
+          <h1 itemProp="headline">{post.frontmatter.title}</h1>
+          <p>{post.frontmatter.date}</p>
+        </header>
+-         {children}
++         <TinaMarkdown content={tinaMarkdownContent} />
+        <hr />
+  // ...
+  )
+}
+```
+
+Unfortunately, we'll need to move images to our media directory to have them appear on our pages. In this case there's only one image
+
+```powershell
+cp ./content/blog/hello-world/salty_egg.jpg static/images/salty_egg.jpg
```

Comment:
Not sure how to make this clearer but I think we should re-word this part as well

Reply:
1. Let me know if you agree with this phrasing

> Tina doesn't currently support the relative image urls used by Gatsby, so we'll need to move our images to the media directory we defined earlier.

2. Should I link the spike regarding this issue? https://github.com/tinacms/tinacms/issues/5172

Reply:
> 1. 

A lot better! 

> 2. 
Possibly.... wouldn't hurt and we could see if people want it 🚀